osutil/epoll/epoll.go

```diff
@@ -116,8 +151,14 @@ func (e *Epoll) Register(fd int, mask Readiness) error {
 //
 // Please refer to epoll_ctl(2) and EPOLL_CTL_DEL for details.
 func (e *Epoll) Deregister(fd int) error {
+	if e.IsClosed() {
+		return ErrEpollClosed
+	}
 	err := unix.EpollCtl(e.fd, unix.EPOLL_CTL_DEL, fd, &unix.EpollEvent{})
 	if err != nil {
+		if e.IsClosed() {
```

Comment:
EBADF, but short of keeping a lock across the calls, there is a very small chance that the descriptor has been reused since already between the check before the call, in that scenario one could get an EINVAL likely. Basically on the contrary there is very little reason to trust the detail of err if Epoll has been closed.

Reply:
Good point, thank you for the explanation.